### PR TITLE
[heat_shaking] split out locking

### PR DESCRIPTION
**Pull request by jkhales**
It's not obvious we always want the plate locked or unlocked when we shake.

### CODE REVIEW

File: pylabrobot/heating_shaking/hamilton.py [hunk 1]
```diff
@@ -61,20 +61,25 @@ async def shake(
...
 
   async def stop_shaking(self):
     """ Shaker `stop_shaking` implementation. """
     await self._stop_shaking()
     await self._wait_for_stop()
-    await self._move_plate_lock(PlateLockPosition.LOCKED)
 
   async def _move_plate_lock(self, position: PlateLockPosition):
+    if not self.lock_initialized:
+      await self._initialize_lock()
```
Comment (rickwierenga) [inline pylabrobot/heating_shaking/hamilton.py:73]: i thought this was only required on `setup`?

Comment (jkhales) [reply pylabrobot/heating_shaking/hamilton.py:73]: We may not want to run it on setup, so this is more flexible, but either way.

Comment (rickwierenga) [reply pylabrobot/heating_shaking/hamilton.py:73]: I generally try to initialize everything on setup to get super predictable protocols

**Comment by jkhales** [comment]
oh nice, was just sitting down to fix this. Thank you for doing it for me!

**Comment by rickwierenga** [comment]
very welcome! thank *you* for contributing this!

**Pull request merged**